virttest/storage.py

```diff
@@ -656,7 +657,34 @@ def __init__(self, params, root_dir, tag):
             self.snapshot_format = ss_params.get("image_format")
 
         self.image_access = ImageAccessInfo.access_info_define_by_params(
-                                                   self.tag, self.params)
+            self.tag, self.params)
+
+        self.data_file = self.external_data_file_defined_by_params(
+            params, root_dir, tag)
+
+    @classmethod
+    def external_data_file_defined_by_params(cls, params, root_dir, tag):
+        """Link image to an external data file."""
+        enable_data_file = params.get("enable_data_file", "no") == "yes"
+        image_format = params.get("image_format", "qcow2")
+        if not enable_data_file:
+            return
+        elif image_format != "qcow2":
+            raise ValueError("The %s format does not support external "
+                             "data file" % image_format)
+        else:
+            image_size = params["image_size"]
+            base_name = os.path.basename(params["image_name"])
+            data_file_path = params.get("data_file_path",
```

Comment:
Suggest the name `image_data_file_path`

Reply:
Since the date_file and base image will be created together with a command \"qemu-img create\", I think no need to set parameter to  image_data_file_path. If I was wrong, please correct me. Many thanks.

e.g.
Create a qcow2: external data file image
# qemu-img create -f qcow2 -o data_file=base.data_file,data_file_raw=off base.qcow2 20G

Reply:
Suppose that other type of devices will have the `data_file` option in the future, there could be name conflicts against the params, so that's why I suggest adding the prefix `image_` as its namespace.

Reply:
updated, thanks.